guides/user/managed-delivery/intergate-your-CI/index.md

```diff
@@ -0,0 +1,151 @@
+---
+layout: single
+title:  "Intergate your CI"
+sidebar:
+  nav: guides
+redirect_from: /reference/managed-delivery/intergate-your-CI
+---
+
+{% include toc %}
+
+One of Managed Delivery goals is to be able to show a code journey, from committing it until it gets deployed,
+in an easy, consistent way.
+In order to enable it, we created an integration between MD and the CI provider, which described below.
+
+## Publishing a new artifact
+
+When a new artifact is published, `keel` gets notified by `echo` (for debians) or by `igor` (for dockers). Each artifact has a `metadata` section, which is controlled by the artifact supplier. 
+For example, for `docker` artifacts, the `metadata` field is being populated in `clouddriver` based on the docker registry. 
+
+Here is an example of a docker artifact event, that `keel` is tracking:
+```
+Received artifact published event: ArtifactPublishedEvent(artifacts=[PublishedArtifact(name=mce/agent, type=DOCKER,
+	reference=dockerRegistry:v2:testregistry:mce/agent, version=h2012.2626f5f, customKind=false, location=testregistry, artifactAccount=null, provenance=null, uuid=null, metadata={date=1607111907158, registry=testregistry, fullname=mce/agent, tag=tag, commitId=2626f5f, buildNumber=2012, branch=tags/v1.2.0^0}
+```
+
+In order to get more metadata per artifact (like build and git details), we created a convension that the `buildNumber` and `commitId` will be part of the artifact metadata object, regardless of the artifact type (it applicable for all types - docker, debians etc.).
+In this example, we will query our CI provider to get metadata for build number `2012` and commit id `2626f5f`.
+
+## Artifact Metadata
+
+The metadata per artifact (like commit message, author, timestamp) is visible in the environment view UI, by clicking on an artifact version.
+
+Here's what it looks like in the UI:
+{%
+  include
+  figure
+  image_path="./artifact-metadata.png"
+%}
+
+In order to get the metadata information, you'll have to plug in your CI provider.
+`keel` is calling `igor` to fetch this information. It uses the endpoint `<igorBaseURL>/ci/builds`, which looks like:
+```
+  @GET("/ci/builds")
+  @Headers("Accept: application/json")
+  suspend fun getArtifactMetadata(
+    @Query("commitId") commitId: String,
+    @Query("buildNumber") buildNumber: String,
+    @Query("projectKey") projectKey: String? = null,
+    @Query("repoSlug") repoSlug: String? = null,
+    @Query("completionStatus") completionStatus: String? = null
+  ): List<Build>?
+```
```

Comment:
You should link to the igor file here

Reply:
and any place where you put the code, I think you should also add the link to the file